### DISCUSSION TITLE
Main Product page variants show ALL, rather than just what's available.

### DISCUSSION THREAD

**Discussion by CodeThirtyTwo** [uicrooks/shopify-foundation-theme#49] [Q&A]
Hey!

I have a question regarding the product variants that display in the <<renderless-product-variant-selector> component.
Currently, it's set to loop through ALL options for all variants and I'm not sure if it's a bug or if it was intended to work this way but my particular use-case has uncovered an issue:

For a particular product, I have one Color, multiple Fit types and multiple Sizes. Example below:

Color: White
Fit Type: Unisex, Ladies, Youth
Sizes: S, M, L, XL, 2XL, 3XL

White/Unisex has all sizes
White/Ladies only has S, M, L
White/Youth only has S, M

In the product variant selector, if I click the dropdown whilst viewing the White/Ladies variant, I still see ALL sizes but when I click XL, 2XL or 3XL it doesn't do anything. These unavailable sizes should be disabled options or not show at all, right?

I've tried to figure out how to do this, but the data that gets passed in doesn't seem to have an easy way of doing this.

Any help here would be greatly appreciated.

**Comment by sergejcodes** [comment]
Hi,

I can replicate this issue. But it seems in Shopify you can only loop through all options of a given product [see here](https://shopify.dev/api/liquid/objects/product#product-options_with_values). The problem you are describing is actually expected behavior. 

I don't think there's a code solution in this case. You can however create all variants and then set the quantity of the variant which you aren't going to use to zero to display a 'sold-out' label for that variant. This is how the official Shopify dawn theme also handles your described case.

![Screenshot 2021-09-28 174945](https://user-images.githubusercontent.com/68379148/135128780-2987f352-2d9e-4a04-9bbb-03c8ea7901db.png)
